### PR TITLE
feat(core): add `init --project-name` CLI option

### PR DESCRIPTION
**Pull request by nmussy**
Adds `cdk init --project-name=<name>` CLI option, overriding default `cwd` dirname

Fixes aws/aws-cdk-cli#176

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/init.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const CDK_HOME = process.env.CDK_HOME ? path.resolve(process.env.CDK_HOME) : pat
 /**
  * Initialize a CDK package in the current directory
  */
-export async function cliInit(type?: string, language?: string, canUseNetwork?: boolean) {
+export async function cliInit(type?: string, language?: string, canUseNetwork = true, projectName?: string) {
```
Comment (eladb) [inline packages/aws-cdk/lib/init.ts]: feels like all these positional arguments should go into a struct...

File: packages/aws-cdk/lib/init.ts [hunk 2]
```diff
@@ -99,7 +101,7 @@ export class InitTemplate {
     const hookTempDirectory = path.join(targetDirectory, 'tmp');
     await fs.mkdir(hookTempDirectory);
     await this.installFiles(sourceDirectory, targetDirectory, {
-      name: decamelize(path.basename(path.resolve(targetDirectory)))
+      name: decamelize(projectName || path.basename(path.resolve(targetDirectory)))
```
Comment (eladb) [inline packages/aws-cdk/lib/init.ts]: do we need to `decamelize` projectName?

**Comment by Jerry-AWS** [comment]
@nmussy nudge? :-) Would really love to use this in the Developer Guide. Got another report today of a problem with Getting Started because the user didn't name his folder the same as ours.

**Comment by nmussy** [comment]
Hey @Jerry-AWS,

I'll look into it some time this week. Sorry for being AWOL for a while there 👋

**Comment by eladb** [comment]
Closing for now

**Pull request closed without merging**